be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/RollCallHandler.scala

```diff
@@ -84,9 +84,10 @@ class RollCallHandler(dbRef: => AskableActorRef) extends MessageHandler {
                 rpcRequest.id
               ))
               case Some(_) =>
+                val laoData: LaoData = readData(rpcRequest.getParamsChannel)
                 val combined = for {
                   _ <- dbActor ? DbActor.ReadLaoData(rpcRequest.getParamsChannel)
-                  _ <- dbActor ? DbActor.WriteLaoData(rpcRequest.getParamsChannel, message)
+                  _ <- dbActor ? DbActor.WriteLaoData(rpcRequest.getParamsChannel, message, laoData.address)
```

Comment:
This is an optional comment but I would avoid an additional DbActor request if possible. A possibility to be able to keep using WriteLaoData without the extra read, but with the additional address modification functionality could be to have an Option with a String (only modify the address in the case Some(str)). This way, you could simply place a None here

Reply:
The approach is not wrong though, just a bit overkill imo

Reply:
You are right. It is safer